app/components/BuyNBTC/SellNBTCTabContent.tsx

```diff
@@ -0,0 +1,163 @@
+import { NBTCIcon, SUIIcon } from "../icons";
+import { SuiClient } from "@mysten/sui/client";
+import { NumericInput } from "../ui/NumericInput";
+import { parseNBTC } from "~/lib/denoms";
+
+const MIN_NBTC_BALANCE = parseNBTC("0.00002");
+const SUI_AMOUNT_ON_SELL = 0.5;
+
+export function SellNBTCTabContent() {
+	const { toast } = useToast();
+	const client = useSuiClient();
+	const account = useCurrentAccount();
+	const { connectedWallet } = useContext(WalletContext);
+	const isSuiWalletConnected = connectedWallet === ByieldWallet.SuiWallet;
+	const { refetchBalance: refetchNBTCBalance } = useNBTCBalance();
+	const { refetchSUIBalance } = useSuiBalance();
+	const { nbtcOTC } = useNetworkVariables();
+	const {
+		mutate: signAndExecuteTransaction,
+		reset: resetMutation,
+		isPending,
+		isSuccess,
+		isError,
+		data,
+	} = useSignAndExecuteTransaction({
+		execute: async ({ bytes, signature }) =>
+			await client.executeTransactionBlock({
+				transactionBlock: bytes,
+				signature,
+				options: {
+					showObjectChanges: true,
+					showEffects: true,
+					showRawEffects: true,
+				},
+			}),
+	});
+
+	const handleTransaction = useCallback(async () => {
+		if (!account) {
+			console.error("Account is not available. Cannot proceed with the transaction.");
+			toast({
+				title: "Sell nBTC",
+				description: "Account is not available. Cannot proceed with the transaction.",
+				variant: "destructive",
+			});
+			return;
+		}
+		const transaction = await createSellNBTCTxn(account.address, client, nbtcOTC, toast);
+		if (!transaction) {
+			console.error("Failed to create the transaction");
+			return;
+		}
+		signAndExecuteTransaction(
+			{
+				transaction,
+			},
+			{
+				onSettled: () => {
+					refetchSUIBalance();
+					refetchNBTCBalance();
+				},
+			},
+		);
+	}, [account, client, nbtcOTC, signAndExecuteTransaction, toast, refetchSUIBalance, refetchNBTCBalance]);
+
+	const resetForm = useCallback(() => {
+		resetMutation();
+	}, [resetMutation]);
+
+	return (
+		<div className="flex flex-col w-full gap-2">
+			<NumericInput
+				className="h-16"
+				value={Number(NBTC_SELL_PRICE)}
+				readOnly
+				rightAdornments={<NBTCIcon className="mr-5" />}
+			/>
+			<ArrowDown className="text-primary justify-center w-full flex p-0 m-0" />
+			<NumericInput
+				value={SUI_AMOUNT_ON_SELL}
+				readOnly
+				className="h-16"
+				rightAdornments={<SUIIcon className="mr-2" />}
+			/>
+			{isSuiWalletConnected ? (
+				<Button type="button" onClick={handleTransaction} disabled={isPending} isLoading={isPending}>
+					Sell nBTC
+					<ChevronRight />
+				</Button>
+			) : (
+				<SuiModal />
+			)}
+			{(isSuccess || isError) && (
+				<Modal title={"Sell nBTC Transaction Status"} open handleClose={resetForm}>
+					<TransactionStatus
+						isSuccess={data?.effects?.status?.status === "success"}
+						handleRetry={resetForm}
+						txnId={data?.digest}
+					/>
+				</Modal>
+			)}
+		</div>
+	);
+}
```

Comment:
most of this code is copy paste.  Can consolidate it with the "buy" option? You can better organize the code.

Reply:
Yes, I am aware of this. I wanted to get review the transaction logic.